app/kuma-dp/pkg/dataplane/envoy/remote_bootstrap.go

```diff
@@ -109,6 +111,60 @@ func (b *remoteBootstrap) Generate(ctx context.Context, url string, cfg kuma_dp.
 	return envoyBootstrap, &bootstrap.KumaSidecarConfiguration, nil
 }
 
+type UIntOrString struct {
+	Type   string
+	UInt   uint64
+	String string
+}
+
+func maybeReadAsBytes(path string) *UIntOrString {
+	byteContents, err := os.ReadFile(path)
+	if err == nil {
+		contents := strings.TrimSpace(string(byteContents))
+		bytes, err := strconv.ParseUint(contents, 10, 64)
+		if err != nil {
+			return &UIntOrString{
+				Type:   "string",
+				String: contents,
+			}
+		}
+		return &UIntOrString{
+			Type: "int",
+			UInt: bytes,
+		}
+	}
+	return nil
+}
+
+func (b *remoteBootstrap) resourceMetadata(cfg kuma_dp.DataplaneResources) (map[string]string, error) {
+	resources := map[string]string{}
+
+	var maxHeap uint64
+
+	if cfg.MaxHeapSizeBytes == 0 {
+		switch cgroups.Mode() {
+		case cgroups.Legacy:
+			res := maybeReadAsBytes("/sys/fs/cgroup/memory.limit_in_bytes")
+			if res != nil && res.Type == "int" {
+				maxHeap = res.UInt
+			}
+		case cgroups.Hybrid, cgroups.Unified:
+			res := maybeReadAsBytes("/sys/fs/cgroup/memory.max")
+			if res != nil && res.Type == "int" {
+				maxHeap = res.UInt
+			}
+		}
+	} else {
+		maxHeap = cfg.MaxHeapSizeBytes
+	}
+
+	if maxHeap != 0 {
+		resources["max_heap_size_bytes"] = strconv.Itoa(int(0.90 * float64(maxHeap)))
```

Comment:
why 0.9?

Reply:
My thought was that Envoy isn't the only process running here so it shouldn't have _all_ the memory. `0.9` is arbitrary

Reply:
removed this for now